src/table/src/metadata.rs

```diff
@@ -588,6 +593,67 @@ impl TableMeta {
         Ok(meta_builder)
     }
 
+    fn change_column_fulltext(
+        &self,
+        table_name: &str,
+        column_name: &String,
+        options: HashMap<String, String>,
+    ) -> Result<TableMetaBuilder> {
+        let table_schema = &self.schema;
+        let mut meta_builder = self.new_meta_builder();
+
+        let column = &table_schema
+            .column_schema_by_name(column_name)
+            .with_context(|| error::ColumnNotExistsSnafu {
+                column_name,
+                table_name,
+            })?;
+
+        let mut fulltext = if let Ok(Some(f)) = column.fulltext_options() {
+            f
+        } else {
+            FulltextOptions::default()
+        };
+
+        if let Some(f) = parse_fulltext_options(&options) {
+            fulltext = f;
+        }
+
+        let columns: Vec<_> = table_schema
+            .column_schemas()
+            .iter()
+            .cloned()
+            .map(|mut column| {
+                if column.name == *column_name {
+                    column = column
+                        .clone()
+                        .with_fulltext_options(fulltext.clone())
```

Comment:
`ColumnSchema::with_fulltext_options` can take a reference of `FulltextOptions` so that you don't have to clone here.